### DISCUSSION TITLE
Capacitor can't handle Apple App Clips, or?

### DISCUSSION THREAD

**Discussion by felixqueisler** [ionic-team/capacitor#3401] [General]
As most of you are probably aware, Apple announced a technology competing against Android Instant Apps. They announced this @ WWDC2020.

There is a limitation (10MB download) and some more regarding guidelines.

Surely hybrid app developers will be left out in this development and will lose out in the coming playing-field of micro-interactions?

**Comment by ikeith** [comment]
Capacitor can work in an App Clip, as it has access to all the expected SDKs so WKWebView is available. There would definitely be a use-case for a new plugin to support the few [APIs in the AppClip framework](https://developer.apple.com/documentation/app_clips/apactivationpayload) ([and you can discuss that here](https://github.com/capacitor-community/proposals/issues)) but otherwise an App Clip is just an application with a few constraints.

The 10mb limit one of those so you would have to be mindful of how you built the app (which frameworks, dependencies, assets, etc) but there's no real reason a focused experience couldn't fit within that size.

**Reply by asangadev** [reply]
This is interesting! Is that possible to add a new CLI feature with Capacitor 3.0, so we can add an APP CLIP to the main project stricture as a sub hybrid app? Just like we do `npx cap add ios` but something like `npx cap add ios-app-clip` ?

So we can use HTML/CSS/JS rather than SWIFT ?

**Reply by felixqueisler** [reply]
> This is interesting! Is that possible to add a new CLI feature with Capacitor 3.0, so we can add an APP CLIP to the main project stricture as a sub hybrid app? Just like we do `npx cap add ios` but something like `npx cap add ios-app-clip` ?
> 
> So we can use HTML/CSS/JS rather than SWIFT ?

This would make sense if you share code between main project and app clip. Which is very likely to be honest. As a sub-app I mean. Or at least individual parts of the app.

Or you'd have to make a 2nd project and be able to properly build it as an app clip right away, then use libraries on both main app and app clip to share code.

**Comment by PMLS3** [comment]
Anyone know if this is available with Capacitor yet? Or Android instant apps?
